docs/modules/ROOT/pages/upgrades.adoc

```diff
@@ -38,23 +37,24 @@ fn _upgrade(new_class_hash: ClassHash) {
 }
 ----
 
-NOTE: If a contract is deployed without this mechanism, the source code is effectively immutable.
+NOTE: If a contract is deployed without this mechanism, its class hash can still be replaced through {library_calls}.
```

Comment:
👍

Reply:
i wonder if it makes sense to clarify that's not the `library_call` directly but if such call executes a `replace_call`. I'm saying this because the proxy-based upgrades also worked with `library_call` and that could cause some confusion.

otoh maybe it's not worth clarifying, we are already talking about class hash replacements